libs/shared/ofrep-core/src/lib/model/ofrep-api-result.ts

```diff
@@ -0,0 +1,37 @@
+import { EvaluationFailureResponse, EvaluationSuccessResponse } from './evaluation';
+import { BulkEvaluationFailureResponse, BulkEvaluationSuccessResponse } from './bulk-evaluation';
+
+export interface OFREPApiResult<
+  S extends OFREPEvaluationErrorHttpStatus | OFREPEvaluationSuccessHttpStatus,
+  T,
+  R extends Response | undefined = Response | undefined,
+> {
+  readonly httpStatus: S;
+  readonly value: T;
+  readonly httpResponse: R;
+}
+
+export const OFREPEvaluationErrorHttpStatuses = [400, 404, 500] as const;
+export type OFREPEvaluationErrorHttpStatus = (typeof OFREPEvaluationErrorHttpStatuses)[number];
+
+export const OFREPEvaluationSuccessHttpStatuses = [200] as const;
```

Comment:
Should we consider the `304` as a successful response too?

From what I see in the code we will throw a `OFREPApiUnexpectedResponseError` if we receive a 304 in the bulk evaluation.

Reply:
You are right this one was missing.
I will add it later this evening. In this case I would just add an undefined value.

Reply:
Yes perfect 👍

Reply:
@thomaspoignant I added 304 handling as described for bulk evals now.
If the value is 304 the infered type in ts is undefined and this can by handled by the consumer.